### PR TITLE
OKTA-79914

### PR DESCRIPTION
**Pull request by tthompson-okta**
Clarify some assumptions in the Public Client App API, specifically for clients which intend to use the Resource Owner Password flow.

@joelfranusic-okta @karlmcguinness-okta @hassenkaraa-okta @mystiberry-okta @mohitchugh-okta @federations-okta

### CODE REVIEW

File: _source/docs/api/resources/oauth-clients.md [hunk 1]
```diff
@@ -58,22 +61,28 @@ Client applications have the following properties:
...
 | token_endpoint_auth_method | requested authentication method for the token endpoint            | `none`, `client_secret_post`, or `client_secret_basic`                 | FALSE    | FALSE  | FALSE    |
 | _links                     | discoverable resources related to the app                         | [JSON HAL](http://tools.ietf.org/html/draft-kelly-json-hal-06)         | TRUE     | FALSE  | TRUE     | 
 |----------------------------+-------------------------------------------------------------------+------------------------------------------------------------------------+----------+--------+----------|
 
-> `client_id`, `client_id_issued_at`, and `client_secret` are only available after a client is created.
+> The `client_id`, `client_id_issued_at`, and `client_secret` attributes are only available after a client is created.
 
-> Client secret is only shown on the initial creation of a client application (and only if the token_endpoint_auth_method is one that is client-secret-based).
+> Client secret is only shown on the initial creation of a client application (and only if the token_endpoint_auth_method is client-secret-based).
```
Comment (mohitchugh-okta) [inline _source/docs/api/resources/oauth-clients.md]: nit: Do we need the hyphen between client-secret and based?

File: _source/docs/api/resources/oauth-clients.md [hunk 2]
```diff
@@ -58,22 +61,28 @@ Client applications have the following properties:
...
-  endpoints in the OAuth protocol.  However, they are related in that the "grant_types" available to a client influence the "response_types"
-  that the client is allowed to use, and vice versa. For instance, a "grant_types" value that includes "authorization_code" implies a
-  "response_types" value that includes "code", as both values are defined as part of the OAuth 2.0 authorization code grant.
+  endpoints in the [OAuth 2.0 protocol](https://tools.ietf.org/html/rfc6749). However, they are related in that the "grant_types"
+  available to a client influence the "response_types" that the client is allowed to use, and vice versa. For instance, a "grant_types"
+  value that includes "authorization_code" implies a "response_types" value that includes "code", as both values are defined as part of
+  the OAuth 2.0 authorization code grant.
+
+> At least one redirect URI and response type is required for all client types, with one exception: if the client uses the
+  [Resource Owner Password](https://tools.ietf.org/html/rfc6749#section-4.3) flow (e.g. if `grant_types` contains the value `password`)
```
Comment (mohitchugh-okta) [inline _source/docs/api/resources/oauth-clients.md]: i.e. instead of e.g. as this is the only use-case where you don't need redirect URI or response_Type for now.

**Comment by mohitchugh-okta** [comment]
Looks good. Ship it!

**Comment by joelfranusic-okta** [comment]
@tthompson-okta - this looks good me me, modulo @mohitchugh-okta's nits.

File: _source/docs/api/resources/oauth-clients.md (hunk 1 above)
Comment (mystiberry-okta) [inline _source/docs/api/resources/oauth-clients.md]: "client-server based" is as hard to read & understand as with the hyphen. I don't know what it means for a method to be "based on client-server", so I can't rewrite accurately, but wondering if we mean "and only if the token_endpoint_auth_method is executed in a client-server environment"?

File: _source/docs/api/resources/oauth-clients.md (hunk 2 above)
Comment (mystiberry-okta) [inline _source/docs/api/resources/oauth-clients.md]: We try to avoid Latin abbreviations altogether. You could simply delete "e.g. if "

Comment (mystiberry-okta) [inline _source/docs/api/resources/oauth-clients.md]: We = tech writers. Didn't mean to invoke the Royal We :)

File: _source/docs/api/resources/oauth-clients.md (hunk 1 above)
Comment (tthompson-okta) [inline _source/docs/api/resources/oauth-clients.md]: I'll take the extra hyphen out, and reword it to say `..(and only if the token_endpoint_auth_method is one that requires a client secret)`.

Comment (mystiberry-okta) [inline _source/docs/api/resources/oauth-clients.md]: fabulous!


________________________________
